src/clj/rems/api/services/util.clj

```diff
@@ -18,46 +17,39 @@
 (defn check-allowed-organization! [organization]
   (assert (:organization/id organization) {:error "invalid organization"
                                            :organization organization})
-  (when (forbidden-organization? organization)
+  (when (forbidden-organization? (organizations/get-organization-raw organization))
     (throw-forbidden (str "no access to organization " (pr-str (:organization/id organization))))))
 
 (deftest test-forbidden-organization?
   (testing "for owner, all organizations are permitted"
-    (binding [context/*user* {:organizations [{:organization/id "own organization"}]}
+    (binding [context/*user* {:eppn "x"}
               context/*roles* #{:owner}]
       (is (not (forbidden-organization? {:organization/id "own organization"})))
       (is (not (forbidden-organization? {:organization/id "not own organization"})))
       (is (not (forbidden-organization? {:organization/id ""})))
       (is (not (forbidden-organization? {:organization/id nil})))))
 
   (testing "for handler, all organizations are permitted"
-    (binding [context/*user* {:organizations [{:organization/id "own organization"}]}
+    (binding [context/*user* {:eppn "x"}
               context/*roles* #{:handler}]
       (is (not (forbidden-organization? {:organization/id "own organization"})))
       (is (not (forbidden-organization? {:organization/id "not own organization"})))
       (is (not (forbidden-organization? {:organization/id ""})))
       (is (not (forbidden-organization? {:organization/id nil})))))
 
   (testing "for owner who is also an organization owner, all organizations are permitted"
-    (binding [context/*user* {:organizations [{:organization/id "own organization"}]}
-              context/*roles* #{:owner :organization-owner}]
+    (binding [context/*user* {:eppn "x"}
+              context/*roles* #{:owner}]
       (is (not (forbidden-organization? {:organization/id "own organization"})))
-      (is (not (forbidden-organization? {:organization/id "not own organization"})))
+      (is (not (forbidden-organization? {:organization/id "not own organization" :organization/owners [{:userid "x"}]})))
```

Comment:
Could be made slightly more obvious if the id would be "organization owned by" or something?